sparse_autoencoder/train/metrics/metric_class.py

```diff
@@ -0,0 +1,30 @@
+"""Base class for metrics."""
+from abc import ABC, abstractmethod
+from typing import TypedDict
+
+from jaxtyping import Float
+from torch import Tensor
+from torch.optim import Optimizer
+
+from sparse_autoencoder.autoencoder.model import SparseAutoencoder
+
+
+class MetricArgs(TypedDict):
+    """Class to hold arguments to metrics, contains everything needed to compute any metric."""
+
+    step: int
+    batch: Float[Tensor, "batch input_activations"]
+    reconstruction_loss_mse: Float[Tensor, " item"]
+    l1_loss: Float[Tensor, " item"]
+    autoencoder: SparseAutoencoder
+    optimizer: Optimizer
+    learned_activations: Float[Tensor, "batch learned_activations"]
+    reconstructed_activations: Float[Tensor, "batch input_activations"]
```

Comment:
Great to have a typed dict for this, as it means we can extend the number of parameters passed without breaking peoples existing loss functions. I think we should add this part in (noting I've got a base metric class I'll probably have merged in by the time you read this so I think we can just update it)

However we shouldn't be passing the autoencoder/optimizer in as this couples these classes with the loss classes and makes it hard to reason about the very different objects. Instead we're using a mediator pattern (aka a pipeline) so the mediator should get any data we need from these (e.g. the current weights from the model) and pass them in directly to the loss functions.